lib/widgets/home.dart

```diff
@@ -2,96 +2,489 @@ import 'package:flutter/material.dart';
 import 'message_list.dart';
 import 'page.dart';
+import 'profile.dart';
 import 'recent_dm_conversations.dart';
 import 'store.dart';
 import 'subscription_list.dart';
 import 'text.dart';
+import 'theme.dart';
 
-class HomePage extends StatelessWidget {
-  const HomePage({super.key});
+enum HomePageTab {
+  inbox,
+  channels,
+  directMessages,
+}
+
+typedef NavigationButtonBuilder = Widget Function(HomePageTab tab);
+
+class HomePage extends StatefulWidget {
+  const HomePage({super.key, required this.initialTab});
 
-  static Route<void> buildRoute({required int accountId}) {
+  static Route<void> buildRoute({required int accountId, HomePageTab? initialTab}) {
     return MaterialAccountWidgetRoute(accountId: accountId,
-        page: const HomePage());
+      page: HomePage(initialTab: initialTab ?? HomePageTab.inbox));
   }
 
+  final HomePageTab initialTab;
+
   @override
-  Widget build(BuildContext context) {
-    final store = PerAccountStoreWidget.of(context);
-    final zulipLocalizations = ZulipLocalizations.of(context);
+  State<HomePage> createState() => _HomePageState();
+}
 
-    final colorScheme = ColorScheme.of(context);
+class _HomePageState extends State<HomePage> {
+  late final ValueNotifier<HomePageTab> _tab = ValueNotifier(widget.initialTab);
 
-    InlineSpan bold(String text) => TextSpan(
-      style: const TextStyle().merge(weightVariableTextStyle(context, wght: 700)),
-      text: text);
+  @override
+  void initState() {
+    super.initState();
+    _tab.addListener(_tabChanged);
+  }
 
-    int? testStreamId;
-    if (store.connection.realmUrl.origin == 'https://chat.zulip.org') {
-      testStreamId = 7; // i.e. `#test here`; TODO cut this scaffolding hack
+  @override
+  void dispose() {
+    _tab.dispose();
+    super.dispose();
+  }
+
+  void _tabChanged() {
+    setState(() {
+      // The actual state lives in [_tab].
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const pageBodies = [
+      (HomePageTab.inbox,          InboxPageBody()),
+      (HomePageTab.channels,       SubscriptionListPageBody()),
+      // TODO(#1094): Users
+      (HomePageTab.directMessages, RecentDmConversationsPageBody()),
+    ];
+
+    NavigationButton button(HomePageTab tab, IconData icon) {
+      return NavigationButton(icon: icon,
+        selected: _tab.value == tab,
+        onPressed: () {
+          _tab.value = tab;
+        });
     }
 
+    // TODO(a11y): add tooltips for the buttons
+    final navigationButtons = {
+      button(HomePageTab.inbox, ZulipIcons.inbox),
+      button(HomePageTab.channels, ZulipIcons.hash_italic),
+      // TODO(#1094): Users
+      button(HomePageTab.directMessages, ZulipIcons.user),
+      NavigationButton(icon: ZulipIcons.menu, selected: false,
+        onPressed: () => _showMenu(context, tabNotifier: _tab)),
+    };
+
+    final designVariables = DesignVariables.of(context);
     return Scaffold(
-      appBar: ZulipAppBar(title: const Text("Home")),
```

Comment:
There's a top bar in the Figma:

<img width="321" alt="image" src="https://github.com/user-attachments/assets/c055a00a-7ba1-40f0-ae39-1dfed37e1a5b">

Ahh but I see #1035 says "There's no top bar" in the issue description. Fair enough 🤷‍♂️ but let's mention it in the commit message's list of differences from the Figma. Also, now we don't have a spec for what goes in the top inset; here's this revision:

| Not scrolled | Scrolled a bit |
| --- | --- |
| <img width="675" alt="image" src="https://github.com/user-attachments/assets/713e959a-2d72-4ed9-859f-66a31c9cfbf5"> | ![image](https://github.com/user-attachments/assets/a4285baa-79ea-4980-9dd4-f5349f37d3b2) |

The top inset is filled with the same color we use under the content in the y-direction when the list is short. I don't love that; it's like the content just abruptly disappears, instead of sliding under some different surface (what used to be the app bar). I don't know what's best here, perhaps just leave a `TODO(design)` for how to fill the top inset on these pages.

Reply:
Yeah, that will be a separate redesign project. I recalled that we should have an issue for the top bar other than #1039, but I couldn't find it.

Reply:
Let's have an app bar at the top after all. (It'll be pretty plain — just a title, I think.)

I'm not sure what my thinking was when I said in the issue there's no top bar. But I see that's a bit of a challenge what should happen on scroll, if there's no app bar; and I just went and looked through several apps on my phone that have a bottom nav bar, and all of them have an app bar on those screens. In a couple of cases the app bar goes away as you scroll (making more room for the scrolling content), but that's trickier; let's skip that at least for launch, and have the app bar remain fixed at the top on scroll.

Reply:
I went to go update #1035 with this change, and rereading the context of that sentence I realized what happened: what I really meant was that there's no top *tab bar*. I probably didn't consider at that moment the question of whether there should or shouldn't be a plain title-bearing app bar instead.

Anyway, fixed there.